MediaBrowser.Providers/Lyric/LrcLyricProvider.cs

```diff
@@ -0,0 +1,189 @@
+
+namespace MediaBrowser.Providers.Lyric;
+
+/// <summary>
+/// LRC Lyric Provider.
+/// </summary>
+public class LrcLyricProvider : ILyricProvider
+{
+    private readonly ILogger<LrcLyricProvider> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LrcLyricProvider"/> class.
+    /// </summary>
+    /// <param name="logger">Instance of the <see cref="ILogger"/> interface.</param>
+    public LrcLyricProvider(ILogger<LrcLyricProvider> logger)
+    {
+        _logger = logger;
+    }
+
+    /// <inheritdoc />
+    public string Name => "LrcLyricProvider";
+
+    /// <summary>
+    /// Gets the priority.
+    /// </summary>
+    /// <value>The priority.</value>
+    public ResolverPriority Priority => ResolverPriority.First;
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<string> SupportedMediaTypes { get; } = new[] { "lrc", "elrc" };
+
+    /// <summary>
+    /// Opens lyric file for the requested item, and processes it for API return.
+    /// </summary>
+    /// <param name="item">The item to to process.</param>
+    /// <returns>If provider can determine lyrics, returns a <see cref="LyricResponse"/> with or without metadata; otherwise, null.</returns>
+    public LyricResponse? GetLyrics(BaseItem item)
+    {
+        string? lyricFilePath = LyricInfo.GetLyricFilePath(this, item.Path);
+
+        if (string.IsNullOrEmpty(lyricFilePath))
+        {
+            return null;
+        }
+
+        IDictionary<string, string> fileMetaData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        string lrcFileContent = System.IO.File.ReadAllText(lyricFilePath);
+
+        Song lyricData;
+
+        try
+        {
+            // Parse and sort lyric rows
+            LyricParser lrcLyricParser = new LrcParser.Parser.Lrc.LrcParser();
+            lyricData = lrcLyricParser.Decode(lrcFileContent);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error parsing lyric data from {Provider}", Name);
+            return null;
+        }
+
+        List<LrcParser.Model.Lyric> sortedLyricData = lyricData.Lyrics.Where(x => x.TimeTags.Count > 0).OrderBy(x => x.TimeTags.First().Value).ToList();
+
+        // Parse metadata rows
+        var metaDataRows = lyricData.Lyrics
+            .Where(x => x.TimeTags.Count == 0)
+            .Where(x => x.Text.StartsWith('[') && x.Text.EndsWith(']'))
+            .Select(x => x.Text)
+            .ToList();
+
+        foreach (string metaDataRow in metaDataRows)
+        {
+            int colonCount = metaDataRow.Count(f => (f == ':'));
+            if (colonCount == 0)
+            {
+                continue;
+            }
+
+            string[] metaDataField = metaDataRow.Split(':', 2, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            string metaDataFieldName = metaDataField[0][1..].Trim();
+            string metaDataFieldValue = metaDataField[1][..^1].Trim();
```

Comment:
```suggestion
            string metaDataFieldName = metaDataField[0][1..];
            string metaDataFieldValue = metaDataField[1][..^1];
```

Now that trim is handled by the split